src/main/java/com/jarslab/maven/babel/plugin/BabelMojo.java

```diff
@@ -63,100 +66,51 @@ public void execute() throws MojoExecutionException, MojoFailureException
             return;
         }
 
-        final SourceFilesExtractor sourceFilesExtractor = new SourceFilesExtractor(
-                sourceDir, jsSourceFiles, jsSourceIncludes, jsSourceExcludes);
-        final TargetFileWriter targetFileWriter = new TargetFileWriter(sourceDir, targetDir, prefix, charset);
-        final String formattedPresets = getFormattedPresets(presets);
-        final Set<Path> sourceFiles = sourceFilesExtractor.getSourceFiles();
+        final TranspilationInitializer transpilationInitializer = TranspilationInitializer.builder()
+                .sourceDirectory(sourceDir.toPath())
+                .targetDirectory(targetDir.toPath())
+                .prefix(prefix)
+                .files(jsSourceFiles)
+                .includes(jsSourceIncludes)
+                .excludes(jsSourceExcludes)
+                .build();
+
+        final Set<TranspileContext> transpilations = transpilationInitializer.getTranspilations();
+
+        if (transpilations.isEmpty()) {
+            getLog().info("No files found to transpile.");
+            return;
+        }
+
         if (verbose) {
-            getLog().info(String.format("Found %s files to transpile.", sourceFiles.size()));
+            getLog().info(format("Found %s files to transpile.", transpilations.size()));
         }
+
+        final TargetFileWriter targetFileWriter = new TargetFileWriter(charset, getLog());
+
+        BabelTranspiler transpiler = BabelTranspiler.builder()
+                .verbose(verbose)
+                .log(getLog())
+                .babelSource(babelSrc)
+                .presets(getFormattedPresets(presets))
+                .charset(charset)
+                .build();
+
         try {
-            final Stream<BabelTranspiler> transpilers = sourceFiles.stream()
-                    .map(sourceFile -> new BabelTranspiler(
-                            verbose,
-                            getLog(),
-                            targetFileWriter,
-                            babelSrcFile,
-                            sourceFile,
-                            formattedPresets,
-                            charset));
-            if (parallel) {
-                getLog().info("Run in parallel mode.");
-                transpilers.parallel();
-            }
-            transpilers
-                    .peek(this::logExtractedFile)
-                    .forEach(BabelTranspiler::execute);
+            transpilations.parallelStream()
+                    .map(transpiler::execute)
+                    .forEach(targetFileWriter::writeTargetFile);
```

Comment:
I see I've dropped the parallel processing mode here, which hasn't been quite intentional. I'l have to revert that. At least until I adopt reusing the `ScriptEngine` (`ReuseEngineTest#testReuseEngine`) in which case parallel transpiling is no longer an option (but it's much faster). 

- [ ] Make parallel processing optional again

Reply:
This has been addressed in #12

Reply:
Makes sense (BTW I like the thread option idea), could we merge PR one after another? (I will postpone releasing new version until then).

Reply:
Sure, no problem. I'll see if I can find some time tonight to work on #12